src/linux_usbmon.pyx

```diff
@@ -19,30 +22,183 @@ from posix.mman cimport mmap, munmap, MAP_FAILED, MAP_PRIVATE, PROT_READ
 MON_IOCQ_RING_SIZE = 37381
 MON_IOCX_MFETCH = -1072655865
 
-ctypedef struct mon_fetch_arg:
-  uint32_t *offvec
-  uint32_t nfetch
-  uint32_t nflush
-
-
-ctypedef struct usbmon_packet:
-  uint64_t id
-  uint8_t type
-  uint8_t xfer_type
-  uint8_t epnum
-  uint8_t devnum
-  uint16_t busnum
-  int8_t flag_steu
-  int8_t flag_data
-  int64_t ts_sec
-  int32_t ts_usec
-  int32_t status
-  uint32_t length
-  uint32_t len_cap
-  uint8_t setup[8]
-  int32_t interval
-  int32_t start_frame
-  int32_t ndesc
+USBMON_PACKET_FIELDS_NAMES = (
+  'id',
+  'type',
+  'xfer_type',
+  'epnum',
+  'devnum',
+  'busnum',
+  'flag_steu',
+  'flag_data',
+  'ts_sec',
+  'ts_usec',
+  'status',
+  'length',
+  'len_cap',
+  'setup',
+  'interval',
+  'start_frame',
+  'xfer_flags',
+  'ndesc',
+)
+
+
+cdef class UsbmonPkt:
+    cdef usbmon_packet *_ptr
+    cdef bint ptr_owner
+
+    def __cinit__(self):
+        self.ptr_owner = False
+
+    def __dealloc__(self):
+        if self._ptr is not NULL and self.ptr_owner is True:
+            free(self._ptr)
+            self._ptr = NULL
+
+    def __str__(self):
+        """Short info"""
+        return f"<usbmon a:{self.address} id:{self.id:08x} p:{self.length}B at {hex(id(self))}>"
+
+    def __repr__(self):
+        """Repr alike in dataclasses, but with positional"""
+        args = [f'{str(getattr(self, name))}' for name in USBMON_PACKET_FIELDS_NAMES]
```

Comment:
You don't need to use `str()` when using f-strings because the conversion to string is the default.

So you could write this as either `f'{getattr(self, name)}'` or avoid the f-string and use `str(getattr(self, name))` which would be equivalent.